src/App.js

```diff
@@ -1,18 +1,113 @@
-import React, { useState } from "react";
-// import React, { Component } from "react";
+import React, { useState, useEffect } from "react";
 import Score from "./components/Score";
 import { Board } from "./components/Board";
 import "./styles.css";
-// import score
 import { ScoreProvider } from "./scoreContext";
-let numCards = 12;
+import { CARD_STATE, CARD_DATA, CARD_IMG } from "./card-constants";
+import { nanoid } from "nanoid";
+import randomizeArray from "./helpers/randomizeArray";
+
+// makes number of cards passed in. Returns <Card /> component
+const makeCards = (data) => {
+  let randomArray = randomizeArray(data.length);
+
+  let results = randomArray.map((index) => {
+    return {
+      id: `card-${nanoid()}`,
+      name: data[index].name,
+      frontImg: data[index].front,
+      backImg: CARD_IMG.BACK,
+      blankImg: CARD_IMG.BLANK,
+      showingImg: CARD_IMG.BACK,
+      status: CARD_STATE.FACEDOWN,
+    };
+  });
+
+  return results;
+};
+
 export default function App() {
-  const [score, setScore] = useState(0);
   const [win, setWin] = useState(false);
   const [isGameInitialized, setIsGameInitialized] = useState(false);
   const [appKey, setAppKey] = useState(0);
   const [matchesArray, setMatchesArray] = useState([]);
 
+  //New code
+  const [shuffledCards, setShuffledCards] = useState();
+
+  // New code
+  const flipCardClickHandler = (event, id) => {
+    let updatedCards = cards.map((card) => {
+      if (card.id === id) {
+        card.status = CARD_STATE.FACEUP;
+        card.showingImg = card.frontImg;
+        return card;
+      }
+      return card;
+    });
+
+    setCards(updatedCards);
+  };
+
+  const flipCardDown = () => {
+    let updatedCards = cards.map((card) => {
+      if (card.status === CARD_STATE.FACEUP) {
```

Comment:
you should be checking for the `card.id` of the array that will be passed through as an argument and essentially flipping those `status` to `CARD_STATE.FACEDOWN`